src/linear-client.ts

```diff
@@ -0,0 +1,356 @@
+import { GenericAPIResponse, getRestBaseUrl, RestClientOptions } from './util/requestUtils';
+import RequestWrapper from './util/requestWrapper';
+import SharedEndpoints from './shared-endpoints';
+
+export class LinearClient extends SharedEndpoints {
+  protected requestWrapper: RequestWrapper;
+
+  /**
+   * @public Creates an instance of the inverse REST API client.
+   *
+   * @param {string} key - your API key
+   * @param {string} secret - your API secret
+   * @param {boolean} [useLivenet=false]
+   * @param {RestClientOptions} [restInverseOptions={}] options to configure REST API connectivity
+   * @param {AxiosRequestConfig} [requestOptions={}] HTTP networking options for axios
+   */
+  constructor(
+    key?: string | undefined,
+    secret?: string | undefined,
+    useLivenet?: boolean,
+    restInverseOptions: RestClientOptions = {},
+    requestOptions: AxiosRequestConfig = {}
+  ) {
+    super()
+    this.requestWrapper = new RequestWrapper(
+      key,
+      secret,
+      getRestBaseUrl(useLivenet),
+      restInverseOptions,
+      requestOptions
+    );
+    return this;
+  }
+
+  /**
+   *
+   * Market Data Endpoints
+   *
+   */
+
+  getKline(params: {
+    symbol: string;
+    interval: string;
+    from: number;
+    limit?: number;
+  }): GenericAPIResponse {
+    return this.requestWrapper.get('public/linear/kline', params);
+  }
+
+  getTrades(params: {
+    symbol: string;
+    limit?: number;
+  }): GenericAPIResponse {
+    return this.requestWrapper.get('public/linear/recent-trading-records', params);
+  }
+
+  getLastFundingRate(params: {
+    symbol: string;
+  }): GenericAPIResponse {
+    return this.requestWrapper.get('public/linear/funding/prev-funding-rate', params);
+  }
+
+  getMarkPriceKline(params: {
+    symbol: string;
+    interval: string;
+    from: number;
+    limit?: number;
+  }): GenericAPIResponse {
+    return this.requestWrapper.get('public/linear/mark-price-kline', params);
+  }
+
+  getIndexPriceKline(params: {
+    symbol: string;
+    interval: string;
+    from: number;
+    limit?: number;
+  }): GenericAPIResponse {
+    return this.requestWrapper.get('public/linear/index-price-kline', params);
+  }
+
+  getPremiumIndexKline(params: {
+    symbol: string;
+    interval: string;
+    from: number;
+    limit?: number;
+  }): GenericAPIResponse {
+    return this.requestWrapper.get('public/linear/premium-index-kline', params);
+  }
+
+  /**
+   *
+   * Account Data Endpoints
+   *
+   */
+
+  placeActiveOrder(params: {
+    side: string;
+    symbol: string;
+    order_type: string;
+    qty: number;
+    price?: number;
+    time_in_force: string;
+    take_profit?: number;
+    stop_loss?: number;
+    tp_trigger_by?: string;
+    sl_trigger_by?: string;
+    reduce_only?: boolean;
+    close_on_trigger?: boolean;
+    order_link_id?: string;
+  }): GenericAPIResponse {
+    return this.requestWrapper.post('private/linear/order/create', params);
+   }
```

Comment:
Tiny indent adjustment:
```suggestion
  }): GenericAPIResponse {
    return this.requestWrapper.post('private/linear/order/create', params);
  }
```